presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java

```diff
@@ -268,6 +268,27 @@ public void testTruncate()
         assertFunction("truncate(DECIMAL '1234567890123456789012.999')", createDecimalType(22, 0), SqlDecimal.of("1234567890123456789012"));
         assertFunction("truncate(DECIMAL '-1234567890123456789012.999')", createDecimalType(22, 0), SqlDecimal.of("-1234567890123456789012"));
 
+        // TRUNCATE DOUBLE -> DOUBLE
+        assertFunction("truncate(nan(),-1)", DOUBLE, Double.NaN);
+        assertFunction("truncate(DOUBLE '17.1', -1)", DOUBLE, 10.0);
+        assertFunction("truncate(DOUBLE '1234.56', 1)", DOUBLE, 1234.5);
+        assertFunction("truncate(DOUBLE '1234.56', 0)", DOUBLE, 1234.0);
+        assertFunction("truncate(DOUBLE '-1234.56', 0)", DOUBLE, -1234.0);
+        assertFunction("truncate(DOUBLE '-1234.56', -500)", DOUBLE, 0.0);
+        assertFunction("truncate(DOUBLE '1234.567', 2)", DOUBLE, 1234.56);
+        assertFunction("truncate(DOUBLE '1234.567', 2)", DOUBLE, 1234.56);
+        assertFunction("truncate(DOUBLE '" + Double.MAX_VALUE + "', -408)", DOUBLE, 0.0);
+        assertFunction("truncate(DOUBLE '" + -Double.MAX_VALUE / 10 + "', 408)", DOUBLE, -Double.MAX_VALUE / 10);
+        assertFunction("truncate(DOUBLE '" + (double) Long.MAX_VALUE + "', -15)", DOUBLE, 9.223E18);
+
+        // TRUNCATE REAL -> REAL
+        assertFunction("truncate(REAL '12.333', 0)", REAL, 12.0f);
+        assertFunction("truncate(REAL '-12.333', 0)", REAL, -12.0f);
+        assertFunction("truncate(REAL '12.333', -500)", REAL, 0.0f);
+        assertFunction("truncate(REAL '3.40287e37', -35)", REAL, 3.4E37f);
+        assertFunction("truncate(REAL '3.40287e37', -488)", REAL, 0.0f);
+        assertFunction("truncate(REAL '" + (float) Long.MAX_VALUE + "', -15)", REAL, 9.223E18f);
```

Comment:
Let's add some test cases for infinity and NaN.

Reply:
infinity and NaN cases may land up in DOUBLE case, testcases added